airflow/models.py

```diff
@@ -815,6 +831,16 @@ def current_state(self, session=None):
         return state
 
     @provide_session
+    def get_dagrun(self, session=None):
+        return (
+            session.query(DagRun)
+            .filter_by(
+                dag_id=self.task.dag.dag_id,
+                execution_date=self.execution_date)
+            .first()
```

Comment:
side note fyi: discovered `.one()` recently which is much like `.first()` but raises if not one and only one is returned, which is nice sometimes, not sure about this particular case here